socialhome/streams/app/components/StreamElement.vue

```diff
@@ -106,6 +114,20 @@ export default Vue.component("stream-element", {
                 .then(() => {
                     this.$data.showSharesBox = false
                     this.$data._sharesCount += 1
+                    this.$data._hasShared = true
+                })
+                .catch(err => console.error(err) /* TODO: Proper error handling */)
+        },
+        unshare() {
+            if (!this.isUserAuthenticated) {
+                console.error("Not logged in")
+                return
+            }
+            this.$http.delete(`/api/content/${this.id}/share/`)
+                .then(() => {
+                    this.$data.showSharesBox = false
+                    this.$data._sharesCount -= 1
+                    this.$data._hasShared = false
```

Comment:
Maybe it's time to put all data and logic in a store? Not mandatory for this PR. Just throwing a thought.

Reply:
This is just local state for the component instance, it shouldn't belong to the state imho?